### DISCUSSION TITLE
Is `record.set(value)` thread safe?

### DISCUSSION THREAD

**Discussion by coretl** [dls-controls/pythonSoftIOC#143] [Q&A]
@evalott100 needs to set a PandA record from another OS thread (file writing, hence OS thread rather than coroutine). Is `record.set` threadsafe, or does it need to be done from the same thread each time?

**Comment by Araneidae** [comment]
I would expect `record.set` to be thread safe ... on the whole.  Writing to the same record from multiple threads can result in surprises, but I don't think there is any fundamental constraint here.

Think this ought to be a discussion rather than an issue.
